protocols/renegadex.js

```diff
@@ -0,0 +1,216 @@
+            0, // static,
+            1, // swap
+            2, // random swap
+            3, // shuffle
+            4, // traditional (assign as players connect)
+            5, // traditional + free swap
+            6 // ladder rank
+          ]
+        },
+        'Game Type': {
+          type: 'integer',
+          description: 'Type of the game the server is running',
+          enum: [
+            0, // Rx_Game_MainMenu
+            1, // Rx_Game
+            2, // TS_Game
+            3 // SP_Game
+            // < 3 x < 1000 = RenX Unused/Reserved
+            // < 1000 < x < 2^31 - 1 = Unassigned / Mod space
+          ]
+        },
+        'Vehicle Limit': {
+          type: 'integer',
+          minimum: 0,
+          description: 'Maximum number of vehicles allowed by this server'
+        },
+        'Mine Limit': {
+          type: 'integer',
+          minimum: 0,
+          description: 'Maximum number of mines allowed by this server'
+        },
+        bPassworded: {
+          type: 'boolean',
+          description: 'Whether a password is required to enter the game'
+        },
+        bSteamRequired: {
+          type: 'boolean',
+          description: 'Whether clients required to be logged into Steam to play on this server'
+        },
+        bRanked: {
+          type: 'boolean',
+          description: 'Whether the serer is ranked/official'
+        },
+        bAllowPrivateMessaging: {
+          type: 'boolean',
+          description: 'Whether the server allows non-admin clients to PM each other'
+        },
+        bPrivateMessageTeamOnly: {
+          type: 'boolean',
+          description: 'whether private messaging is restricted to just teammates'
+        },
+        bAutoBalanceTeams: { // alias of 'bSpawnCrates'
+          type: 'boolean',
+          description: 'Whether the server will spawn crates in this game for balancing'
+        },
+        bSpawnCrates: {
+          type: 'boolean',
+          description: 'Whether the server will spawn crates in this game for balancing'
+        },
+        CrateRespawnAfterPickup: {
+          type: 'integer',
+          minimum: 0,
+          description: 'interval for crate respawn (after pickup)'
+        }
+      },
+      required: [
+        'Player Limit',
+        'Time Limit',
+        'Team Mode',
+        'Game Type',
+        'Vehicle Limit',
+        'Mine Limit'
+      ]
+    }
+  }
+}
+export const MasterServerResponseSchema = {
+  type: 'array',
+  items: { $ref: '#/$defs/server' },
+  $defs: {
+    server: MasterServerServerInfoSchema
+  }
+}
+
+/**
+ * Implements the protocol for Renegade X, an UnrealEngine3 based game, using a custom master server
+ */
+export default class renegadex extends Core {
+  constructor () {
+    super()
+    this.usedTcp = true
+  }
+
+  async run (state) {
+    const servers = await this.request({
+      url: 'https://serverlist-rx.totemarts.services/servers.jsp',
+      responseType: 'json'
+    })
```

Comment:
Please split this protocol to a master one (`renegadexmaster`) and this one (`renegadex`) to let users to query just the master server to obtain the server list if they'd like to.

Other similar cases of this: 
* [beammpmaster](https://github.com/gamedig/node-gamedig/blob/c77ca299b017619506bdb0680e5809e36364520f/protocols/beammpmaster.js) - [beammp](https://github.com/gamedig/node-gamedig/blob/c77ca299b017619506bdb0680e5809e36364520f/protocols/beammp.js)
* [vintagestorymaster](https://github.com/gamedig/node-gamedig/blob/c77ca299b017619506bdb0680e5809e36364520f/protocols/vintagestorymaster.js) - [vintagestory](https://github.com/gamedig/node-gamedig/blob/c77ca299b017619506bdb0680e5809e36364520f/protocols/vintagestory.js)
* [tribes1master](https://github.com/gamedig/node-gamedig/blob/c77ca299b017619506bdb0680e5809e36364520f/protocols/tribes1master.js) - note that [tribes1](https://github.com/gamedig/node-gamedig/blob/c77ca299b017619506bdb0680e5809e36364520f/protocols/tribes1.js) does not use the master server for the query, but we still have the master server implementation.

Reply:
That master-separation doesn't seem to be fully implemented (respectively exposed for the API). The master query logic is also a bit inconsistent throughout the protocols (see `altvml` and `ragemp` [or `minetest`]).

Also by having a master protocol (like `vintagestorymaster`) there could be the assumption of receiving a list of processed server data and not just the raw master server data. And vice versa

Not sure how I feel about that. Should I still follow the mentioned split-master-logic?

Reply:
I'd say yes, although we don't really mention anything about the support of querying master servers (and that it is flaky right now) we can provide it as just it is right now and straighten it up for the next major release (we should do a documented issue about this).

Reply:
Added that master protocol. I can't tell if I like the current logic of adding master-protocols (the earlier subsystem I propsoed in #638 could be a alternative), but here we go.

The master server can be used wih `protocol-renegadexmaster` (or to be consistent to current game ID naming, `protocol-renegade10master`. The server list will be processed to not provide different type of data.